src/Components/Patient/DailyRounds.tsx

```diff
@@ -492,6 +489,17 @@ export const DailyRounds = (props: any) => {
     dispatch({ type: "set_form", form });
   };
 
+  function handleLocalTemperatureChange(e: any) {
+    if (e.key === "temperature") {
+      if (e.oldValue === "C" && e.newValue === "F") {
+        toggleTemperature();
+      } else if (e.oldValue === "F" && e.newValue === "C") {
+        toggleTemperature();
+      }
+    }
+  }
+  window.addEventListener("storage", handleLocalTemperatureChange);
```

Comment:
Shouldn't you be wrapping this listener inside a `useEffect` and dispose the listener when the component is unmounted?

Reply:
I hadn't worked with this localstorage earlier. Thanks for pointing it out, now I am a bit familiar with localstorage methods.